.github/workflows/smoke-tests.yml

```diff
@@ -0,0 +1,90 @@
+name: smoke-tests
+run-name: Smoke Tests
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 3 * * 1-5'
+
+# limit the access of the generated GITHUB_TOKEN
+permissions:
+  contents: read
+
+env:
+  AWS_ACCOUNT_SECRET: 'secret/observability-team/ci/elastic-observability-aws-account-auth'
+  EC_KEY_SECRET: 'secret/observability-team/ci/elastic-cloud/observability-pro'
+
+jobs:
+  prepare:
+    name: Generate smoke tests list
+    runs-on: ubuntu-latest
+    outputs:
+      tests: ${{ steps.generate.outputs.tests }}
+      date: ${{ steps.generate.outputs.date }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup cluster env
+        uses: ./.github/workflows/setup-cluster-env
+        with:
+          vault-url: ${{ secrets.VAULT_ADDR }}
+          vault-role-id: ${{ secrets.VAULT_ROLE_ID }}
+          vault-secret-id: ${{ secrets.VAULT_SECRET_ID }}
+      - id: generate
+        name: Generate matrix and date
+        run: |
+          echo "tests=$(make smoketest/discover)" >> "${GITHUB_OUTPUT}"
+          echo "date=$(date +%s)" >> "${GITHUB_OUTPUT}"
+
+  smoke-tests-ess:
+    name: Run smoke tests ${{ matrix.test }} for ${{ matrix.version }}
+    runs-on: ubuntu-latest
+    needs: prepare
+    env:
+      TF_VAR_BUILD_ID: ${{ github.run_id }}
+      TF_VAR_ENVIRONMENT: 'ci'
+      TF_VAR_BRANCH: ${{ github.ref_name }}
+      TF_VAR_REPO: ${{ github.repository }}
+      TF_VAR_CREATED_DATE: ${{ needs.prepare.outputs.date }}
+    strategy:
+      matrix:
+        test: ${{ fromJSON(needs.prepare.outputs.tests) }}
+        version:
+          - '7.17'
+          - 'latest'
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup cluster env
+        uses: ./.github/workflows/setup-cluster-env
+        with:
+          vault-url: ${{ secrets.VAULT_ADDR }}
+          vault-role-id: ${{ secrets.VAULT_ROLE_ID }}
+          vault-secret-id: ${{ secrets.VAULT_SECRET_ID }}
+      - name: Run smoke tests ${{ matrix.test }} for ${{ matrix.version }}
+        run: make smoketest/run-version TEST_DIR=${{ matrix.test }} SMOKETEST_VERSION=${{ matrix.version }}
+      - if: always()
+        name: Teardown smoke test infra
+        run: make smoketest/cleanup TEST_DIR=${{ matrix.test }}
+
+  smoke-tests-os:
```

Comment:
These tests are currently run for `'main', '8.<minor>', '8.<next-patch>'` (see 
[def](https://github.com/elastic/apm-server/pull/10654/files#diff-74ea6b734b6243d4e4e4dddde4234041d59d005d03fbc87f7265d32065a73df2L24)); where is this defined in the github worfklow?

Reply:
I checked the current [behavior](https://apm-ci.elastic.co/job/apm-server/job/smoke-tests-os-mbp/ ) and currently only the `main` branch is scheduled.
It isn't scheduling `8.x` due to the missing `.ci/smoke-tests-os.groovy` Jenkins script.
Does the list above explicit? Do we need to run `Smoke Tests OS` on `main`, `8.7` and `8.7.1` branches?

Reply:
oh interesting; yes it would be good to run the smoke tests on these branches; 
`'8.<minor>', '8.<next-patch>'` make more sense when we are in FF phase, e.g. once we branch out `8.8` but before releasing it we would run the smoke tests on `main` (`8.9.0`), `8.8.0` (unreleased), `8.7.1` (next patch version to be released).

Reply:
I replace the script with a GitHub Actions doing the same.